### PR TITLE
Simplify TimeRange.overlaps and TimeRange.contains implementations.

### PR DESCRIPTION
**Pull request by GrantComm**
Removed some redundant booleans in the implementation of those two functions, suggested by @apluscs.

### CODE REVIEW

**Review by acmcarther — APPROVED** [review]
I recommend a more specific PR title and description. 

When Jack and I look through PRs for final eval, we're going to have to look at this PR's code in order to tell what it's actually doing. A super clear title and description can help us a lot.

Here is how I might have worded this.

```
Simplify TimeRange.overlaps and contains implementations.

This PR removes some redundant booleans in the implementation of those two functions, suggested by @apluscs. Thanks!
```

File: src/main/java/com/google/sps/data/TimeRange.java [hunk 1]
```diff
@@ -79,13 +79,14 @@ public boolean overlaps(TimeRange other) {
...
+    }
+
     if (this.start.isBefore(other.start)) {
       return this.contains(other.start);
     }
 
-    if (this.start.equals(other.start)) {
-      return this.end.isAfter(other.end);
-    }
     return other.contains(this.start);
```
Comment (apluscs) [inline src/main/java/com/google/sps/data/TimeRange.java]: This function looks pretty good, but can be shortened if you think about what cases would mean no overlap.
There's only no overlap when other.end <= this.start OR other.start >= this.end. So in code:
``` ! (!other.end.isAfter(this.start) || !other.start.isBefore(this.end)) ``` = 
``` other.end.isAfter(this.start) && other.start.isBefore(this.end) ```

**Review by apluscs — CHANGES_REQUESTED** [review]
One logic quibble.
Also, l remember you adding some tests. Are those going to be in another PR?

File: src/main/java/com/google/sps/data/TimeRange.java (hunk 1 above)
Comment (GrantComm) [reply src/main/java/com/google/sps/data/TimeRange.java]: @apluscs Yes, additional test will be in separate pr

**Comment by apluscs** [comment]
Ok since you're adding more tests you should probably put them in the same PR as this one. Then we can know the changes pass these new tests (and also that the tests are more likely to be accurate).

**Review by apluscs — APPROVED** [review]
LGTM

**Pull request merged**